pkg/traceutil/exporter.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceutil
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"go.uber.org/zap"
+)
+
+// LogExporter writes Span to specified Logger.
+type LogExporter struct {
+	// Log is usually zap.Logger.Info.
+	Log func(msg string, fields ...zap.Field)
+}
+
+var _ trace.SpanExporter = (*LogExporter)(nil)
+
+// NewLogExporter creates a new LogExporter which will write Spans as Log messages.
+func NewLogExporter(logger *zap.Logger) *LogExporter {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	return &LogExporter{Log: logger.Info}
+}
+
+func (e *LogExporter) ExportSpans(ctx context.Context, spans []trace.ReadOnlySpan) error {
+	for _, span := range spans {
+		msg, fields := logSpan(span)
+		e.Log(msg, fields...)
+	}
+	return nil
+}
+
+func (e *LogExporter) Shutdown(ctx context.Context) error {
+	return nil
+}
+
+func logSpan(s trace.ReadOnlySpan) (string, []zap.Field) {
+	start := s.StartTime()
+	end := s.EndTime()
+	duration := end.Sub(start)
+	events := s.Events()
+	steps := make([]string, 0, len(events))
+	slices.SortFunc(events, func(a, b trace.Event) int {
```

Comment:
How does it work with nesting?

Reply:
From PR description:
> Currently, the only difference are "child traces" which are not yet instrumented in Etcd's OpenTelemetry tracing. This can be later mitigated by adding "links" or copying events between spans. I also have other ideas...

so either printing `spanID` so it will be easy to search in logs/Otel collector or returning the span to use for copying like in https://github.com/etcd-io/etcd/pull/20307